### ISSUE TITLE
Documentation lacking for accessing remote hosts

### ISSUE THREAD

**Issue by candlerb** [lxc/pylxd#79]
pylxd documentation doesn't say how to access remote hosts. Poking around, I find a number of issues.

1. The [front page](https://github.com/lxc/pylxd) intro document says to use `pylxd.api.API()`, however in the code pylxd.api is imported from pylxd.deprecated. Should anyone even be using this in the first place??

    Looking at [ansible lxd_container](https://github.com/ansible/ansible-modules-extras/pull/2208), they use `pylxd.client.Client()` instead. I haven't found any documentation for that entry point into pylxd. Indeed, if there is any documentation for pylxd, I've been unable to find it: only a skeleton at <https://github.com/lxc/pylxd/tree/master/doc/source>

2. From the source, `pylxd.api.API()` takes an optional host and port. I found that I can use `pylxd.api.API(host="nuc1.ws.nsrc.org").container_list()` and this works just fine. It doesn't matter which variation of the host's name I use.

3. From the source, `pylxd.client.Client()` takes an optional URL. However if I try this code:

    ~~~
    pylxd.client.Client('https://nuc1w.ws.nsrc.org:8443').containers.all()
    ~~~

    then I get:

    ~~~
    requests.exceptions.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)
    ~~~

    I have tried all the variations of the host name, and also connecting by IP address. If I connect using `openssl s_client` then I see the certificate has:

    ~~~
            Subject: O=linuxcontainers.org, CN=root@nuc1.ws.nsrc.org
    ...
                X509v3 Subject Alternative Name:
                    DNS:nuc1.ws.nsrc.org, DNS:fe80::c23f:d5ff:fe63:6411/64, DNS:10.10.0.238/24, DNS:10.10.0.237/32, DNS:2001:db8:100::238/64, DNS:fe80::c23f:d5ff:fe63:6411/64, DNS:fe80::fcd5:94ff:fe50:96ca/64, DNS:fe80::8461:ddff:fe6f:3aa8/64, DNS:fe80::1/64
    ~~~

    i.e. all hostname variations are in SAN (except I don't know why the certificate would include the prefix length, but then I've never seen certificates with IP addresses in them)

So anyway: for code which uses pylxd.client.Client() then I'd like to see an example of how to make it work without certificate errors.  I'd also like it to be clear whether pylxd.api or pylxd.client is the recommended interface to use.

**Comment by jpic** [comment]
4d4517c54862fedd4a725aebcd628d7ddc0279d5 adds missing code and test for remote connection.

I'd like to volunteer to write the docs, if repo maintainers would be ok with that, even though I'm not a native english speaker, I <3 writing docs + it's a great way to learn new stuff !

**Comment by candlerb** [comment]
I don't think anyone will refuse an offer of having docs written 👍 

What I don't understand is how `Client()` was ever supposed to work to access a remote host, if you can't pass in a key and cert. For testing you have buried some environment variable lookups, but that's not useful if you want to access more than one host in your application.

Wouldn't they need to be exposed as arguments on `Client.__init__` ? Or else I'm missing something.

**Comment by jpic** [comment]
Correct, for now I'm just trying to have some basic CI up and running again before I propose any change that could break someone else's code.

Currently, I couldn't find how to build the docs - and I'm not sure how the include of README.md in readme.rst would work. Perhaps @zulcss would know, otherwise I can start from a fresh sphinx-quickstart.

**Comment by jpic** [comment]
Note that my branch now exposes `client_{crt,key}` as well as `insecure` as Client constructor arguments.

**Comment by rockstar** [comment]
`Client` was initially developed to work against the unix socket, not the http interface. Clearly, this was a use-based thing. If someone puts a patch up before I get around to it, I'd be happy to review it and help it get merged (particularly because our current use case doesn't cover this).

**Comment by rockstar** [comment]
Fixed in 0d55ade
